app/src/main/java/org/simple/clinic/newentry/PatientEntryInit.kt

```diff
@@ -6,5 +6,5 @@ import com.spotify.mobius.Init
 
 class PatientEntryInit : Init<PatientEntryModel, PatientEntryEffect> {
   override fun init(model: PatientEntryModel): First<PatientEntryModel, PatientEntryEffect> =
-      first(model, setOf(FetchPatientEntry, LoadInputFields))
+      first(model, setOf(FetchPatientEntry, LoadInputFields, FetchColonyOrVillagesEffect))
```

Comment:
Let's add a check to make sure we are not fetching the list again when village or colony names are already present in the model.

Reply:
Can you suggest a better way to achieve this?
 ```
    val effects = setOf(
        PrefillFormEffect(patient, address, phoneNumber, bangladeshNationalId),
        FetchBpPassportsEffect(patient.uuid),
        LoadInputFields
    )
    
    val effect1 = mutableSetOf(
        PrefillFormEffect(patient, address, phoneNumber, bangladeshNationalId),
        FetchBpPassportsEffect(patient.uuid),
        LoadInputFields,
        FetchColonyOrVillagesEffect
    )

    return if (model.hasColonyOrVillagesList) {
      first(model, effects)
    } else {
      first(model, effect1)
    }
```

Reply:
We can use a mutable set, so we can do something like this

```kt
val effects = mutableSetOf(Effect1, Effect2)

if (model.hasColonyOrVillagesList) {
  effects.add(FetchEffect)
}

first(model, effects)
```